basement/Basement/UTF8/Base.hs

```diff
@@ -104,12 +104,12 @@ sToList (String arr) = Vec.onBackend onBA onAddr arr
 sToListStream (String arr) k z = Vec.onBackend onBA onAddr arr
   where
     (Vec.ValidRange !start !end) = Vec.offsetsValidRange arr
-    onBA (BLK.Block ba) = loop start
+    onBA ba@(BLK.Block _) = loop start
```

Comment:
is this `ba@(BLK.Block _)` not equivalent to `!ba` ?

Reply:
You are most likely right.

There were other places where a pattern like this (but probably with a bang `!` on the value within the Block) that made a (significant) difference in the produced core. I will address that when combining the "onBA" and "onAddr" in a next step.